### PR TITLE
expect both snake and camel casing for mobile doc upload

### PR DESCRIPTION
**Pull request by aherzberg**
## Summary
 It was found that FE is using both different casing between multi image upload and single image upload. It should be using camel case but for backwards compatibility, we need to handle both. 

see FE code: https://github.com/department-of-veterans-affairs/va-mobile-app/blob/0e42a33022a4fa54692228d07a49e6e7bb003d69/VAMobile/src/store/slices/claimsAndAppealsSlice.ts

## Related issue(s)
https://github.com/department-of-veterans-affairs/va-mobile-app/issues/7154

## Testing done
Specs

## Screenshots
_Note: Optional_

## What areas of the site does it impact?
*(Describe what parts of the site are impacted and*if*code touched other areas)*

## Acceptance criteria

- [ ]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [ ]  No error nor warning in the console.
- [ ]  Events are being sent to the appropriate logging solution
- [ ]  Documentation has been updated (link to documentation)
- [ ]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ]  Feature/bug has a monitor built into Datadog or Grafana (if applicable)
- [ ]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [ ]  I added a screenshot of the developed feature

## Requested Feedback

(OPTIONAL)_What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

**Review by kpethtel — CHANGES_REQUESTED** [review]
The controller changes look fine, but the tests aren't effectively testing the changes. 

By default, the sis headers add `X-Key-Inflection: camel`. The reason it (and the mobile app) does this is because rails developers generally prefer to use snake case while most front ends prefer camel case. This abstracts the format conversion in the middleware and so the front end can send and receive camel case while the back end can send and receive snake case. We do that with the [olive branch gem](https://www.viget.com/articles/introducing-olivebranch/). 

Since these tests all use the x-key-inflection header, the controller receives snake case regardless of what you pass in the specs. I think a better way to test this might be to invert the structure so that you're have a context for testing "when params are camel case". In that context, you'd specify `sis_headers(camelize: false)`, which will allow the params to pass through unmodified. The rest of the tests can be done with default headers. 

On a separate note, do you know why this even matters? The only reason I can think of why this is an issue is that maybe older versions of the mobile app don't specify the x-key-inflection or maybe they just screwed up at some point and didn't specify them when making this call, but that seems odd to me because I'd assume they do it by default. 

The only places we currently use `camelize: false` are the demographics and gender identity specs, and I'm actually not sure why we do it there either. It looks like you wrote the demographics specs and @Tonksthebear wrote the gender identity specs. Any idea why we did that there?

**Comment by aherzberg** [comment]
> The controller changes look fine, but the tests aren't effectively testing the changes.
> 
> By default, the sis headers add `X-Key-Inflection: camel`. The reason it (and the mobile app) does this is because rails developers generally prefer to use snake case while most front ends prefer camel case. This abstracts the format conversion in the middleware and so the front end can send and receive camel case while the back end can send and receive snake case. We do that with the [olive branch gem](https://www.viget.com/articles/introducing-olivebranch/).
> 
> Since these tests all use the x-key-inflection header, the controller receives snake case regardless of what you pass in the specs. I think a better way to test this might be to invert the structure so that you're have a context for testing "when params are camel case". In that context, you'd specify `sis_headers(camelize: false)`, which will allow the params to pass through unmodified. The rest of the tests can be done with default headers.
> 
> On a separate note, do you know why this even matters? The only reason I can think of why this is an issue is that maybe older versions of the mobile app don't specify the x-key-inflection or maybe they just screwed up at some point and didn't specify them when making this call, but that seems odd to me because I'd assume they do it by default.
> 
> The only places we currently use `camelize: false` are the demographics and gender identity specs, and I'm actually not sure why we do it there either. It looks like you wrote the demographics specs and @Tonksthebear wrote the gender identity specs. Any idea why we did that there?

There may be something causing that camelize header not to take effect because when I change the line `      params[:trackedItemId] || params[:tracked_item_id]` to just `params[:trackedItemId]`, only looking for the camel case form, then the snake case specs fail which shouldn't be possible with that camel header

**Comment by kpethtel** [comment]
> > The controller changes look fine, but the tests aren't effectively testing the changes.
> > By default, the sis headers add `X-Key-Inflection: camel`. The reason it (and the mobile app) does this is because rails developers generally prefer to use snake case while most front ends prefer camel case. This abstracts the format conversion in the middleware and so the front end can send and receive camel case while the back end can send and receive snake case. We do that with the [olive branch gem](https://www.viget.com/articles/introducing-olivebranch/).
> > Since these tests all use the x-key-inflection header, the controller receives snake case regardless of what you pass in the specs. I think a better way to test this might be to invert the structure so that you're have a context for testing "when params are camel case". In that context, you'd specify `sis_headers(camelize: false)`, which will allow the params to pass through unmodified. The rest of the tests can be done with default headers.
> > On a separate note, do you know why this even matters? The only reason I can think of why this is an issue is that maybe older versions of the mobile app don't specify the x-key-inflection or maybe they just screwed up at some point and didn't specify them when making this call, but that seems odd to me because I'd assume they do it by default.
> > The only places we currently use `camelize: false` are the demographics and gender identity specs, and I'm actually not sure why we do it there either. It looks like you wrote the demographics specs and @Tonksthebear wrote the gender identity specs. Any idea why we did that there?
> 
> There may be something causing that camelize header not to take effect because when I change the line ` params[:trackedItemId] || params[:tracked_item_id]` to just `params[:trackedItemId]`, only looking for the camel case form, then the snake case specs fail which shouldn't be possible with that camel header

@aherzberg I'm not sure I understand what you mean. The only time you should expect to receive camel case in those specs is if you have `camelize: false` and the params are camel case. The `x-key-inflection: camel` header does not convert the params to camel case in the controller. It ensures that camel case params received by the controller are converted to snake case.

**Comment by kpethtel** [comment]
> > The controller changes look fine, but the tests aren't effectively testing the changes.
> > By default, the sis headers add `X-Key-Inflection: camel`. The reason it (and the mobile app) does this is because rails developers generally prefer to use snake case while most front ends prefer camel case. This abstracts the format conversion in the middleware and so the front end can send and receive camel case while the back end can send and receive snake case. We do that with the [olive branch gem](https://www.viget.com/articles/introducing-olivebranch/).
> > Since these tests all use the x-key-inflection header, the controller receives snake case regardless of what you pass in the specs. I think a better way to test this might be to invert the structure so that you're have a context for testing "when params are camel case". In that context, you'd specify `sis_headers(camelize: false)`, which will allow the params to pass through unmodified. The rest of the tests can be done with default headers.
> > On a separate note, do you know why this even matters? The only reason I can think of why this is an issue is that maybe older versions of the mobile app don't specify the x-key-inflection or maybe they just screwed up at some point and didn't specify them when making this call, but that seems odd to me because I'd assume they do it by default.
> > The only places we currently use `camelize: false` are the demographics and gender identity specs, and I'm actually not sure why we do it there either. It looks like you wrote the demographics specs and @Tonksthebear wrote the gender identity specs. Any idea why we did that there?
> 
> There may be something causing that camelize header not to take effect because when I change the line ` params[:trackedItemId] || params[:tracked_item_id]` to just `params[:trackedItemId]`, only looking for the camel case form, then the snake case specs fail which shouldn't be possible with that camel header

I think one of the problems here is that I named the sis header param `camelize`. I didn't really think about what it was doing when I named it that. It should probably be renamed something like `convert_camel_to_snake` or maybe even just `x_key_inflection: camel`.

**Comment by aherzberg** [comment]
> > > The controller changes look fine, but the tests aren't effectively testing the changes.
> > > By default, the sis headers add `X-Key-Inflection: camel`. The reason it (and the mobile app) does this is because rails developers generally prefer to use snake case while most front ends prefer camel case. This abstracts the format conversion in the middleware and so the front end can send and receive camel case while the back end can send and receive snake case. We do that with the [olive branch gem](https://www.viget.com/articles/introducing-olivebranch/).
> > > Since these tests all use the x-key-inflection header, the controller receives snake case regardless of what you pass in the specs. I think a better way to test this might be to invert the structure so that you're have a context for testing "when params are camel case". In that context, you'd specify `sis_headers(camelize: false)`, which will allow the params to pass through unmodified. The rest of the tests can be done with default headers.
> > > On a separate note, do you know why this even matters? The only reason I can think of why this is an issue is that maybe older versions of the mobile app don't specify the x-key-inflection or maybe they just screwed up at some point and didn't specify them when making this call, but that seems odd to me because I'd assume they do it by default.
> > > The only places we currently use `camelize: false` are the demographics and gender identity specs, and I'm actually not sure why we do it there either. It looks like you wrote the demographics specs and @Tonksthebear wrote the gender identity specs. Any idea why we did that there?
> > 
> > 
> > There may be something causing that camelize header not to take effect because when I change the line ` params[:trackedItemId] || params[:tracked_item_id]` to just `params[:trackedItemId]`, only looking for the camel case form, then the snake case specs fail which shouldn't be possible with that camel header
> 
> I think one of the problems here is that I named the sis header param `camelize`. I didn't really think about what it was doing when I named it that. It should probably be renamed something like `convert_camel_to_snake` or maybe even just `x_key_inflection: camel`.

Oh right. I mixed it up what that header was doing. 

So I verified that at least for this endpoint in question in the current build of the app, they are using the header `x-key-inflection: camel` but, when I set the parameters to use camel case and setup a `binding.pry` in the controller, and leave the SIS headers in the specs untouched so camelize is still true, It's still showing the parameters as camel case. Looking at the request headers in the controller, it's coming out as `HTTP_X_KEY_INFLECTION: camel` which looks correct as there's other parts of vets-api that set that header. So unless I'm still misunderstanding how this header works, I think there's something keeping this header from working in the specs at least. I also just realized that all my postman requests don't use that camel header which I should add for consistency.

**Comment by aherzberg** [comment]
@kpethtel and I paired and found that `x-key-inflection: camel` has no effect on the parameters when the body payload is not json. This is why we're getting different casing coming through even though I have confirmed FE is using that header. making the payload body json in the single doc upload breaks the endpoint so I'm hesitant to change behavior so it can handle json body as it's been done this way even when it was EVSS and FE has probably implemented it in a way without making json. I think the best course right now is to push this change in and see if the problems we're seeing are fixed.

File: modules/mobile/app/controllers/mobile/v0/claims_and_appeals_controller.rb [hunk 1]
```diff
@@ -109,6 +109,9 @@ def set_params
         params.delete(:trackedItemId)
       end
```
Comment (kpethtel) [inline modules/mobile/app/controllers/mobile/v0/claims_and_appeals_controller.rb:111]: Good comment.

File: modules/mobile/spec/request/claims_document_upload_spec.rb [hunk 2]
```diff
@@ -41,53 +41,54 @@
...
 
   it 'uploads multiple gif files' do
     files = [Base64.encode64(File.read('spec/fixtures/files/doctors-note.gif')),
              Base64.encode64(File.read('spec/fixtures/files/marriage-cert.gif'))]
-    params = { files:, claimId: claim_id, trackedItemId: tracked_item_id, documentType: document_type }
+    params = { files:, claim_id:, tracked_item_id:, document_type: }
     expect_any_instance_of(BenefitsDocuments::Service).to receive(:cleanup_after_upload)
+
     expect do
       post '/mobile/v0/claim/600117255/documents/multi-image', params: params.to_json,
```
Comment (kpethtel) [inline modules/mobile/spec/request/claims_document_upload_spec.rb:90]: I guess you could make an argument that we should for the multi-file upload this shouldn't be json to better match reality. But... I also don't think we should be too concerned about that. The front end could change how they do things at any moment, so it's not our role to match how they use the api. We just want to ensure that the code works with both camel and snake.

**Review by kpethtel — APPROVED** [review]
Looks good. Thanks for sticking with it and looking through the front end to figure out what's going on.

**Review by Tonksthebear — APPROVED** [review]
LGTM

**Pull request merged**